docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,586 @@
+            compensations.append("undo_book_car")
+            output = await workflow.execute_activity(
+                book_car,
+                input,
+                start_to_close_timeout=timedelta(seconds=10),
+                retry_policy=RetryPolicy(
+                    non_retryable_error_types=["Exception"],
+                ),
+            )
+
+            # Attempt to book a hotel
+            compensations.append("undo_book_hotel")
+            output += " " + await workflow.execute_activity(
+                book_hotel,
+                input,
+                start_to_close_timeout=timedelta(seconds=10),
+                retry_policy=RetryPolicy(
+                    non_retryable_error_types=["Exception"],
+                ),
+            )
+
+            # Attempt to book a flight
+            compensations.append("undo_book_flight")
+            output += " " + await workflow.execute_activity(
+                book_flight,
+                input,
+                start_to_close_timeout=timedelta(seconds=10),
+                retry_policy=RetryPolicy(
+                    initial_interval=timedelta(seconds=1),
+                    maximum_interval=timedelta(seconds=1),
+                    maximum_attempts=input.attempts,
+                    non_retryable_error_types=["Exception"],
+                ),
+            )
+
+            # If all bookings are successful, return the output
+            return output
+        except Exception:
+            # If an error occurs, execute compensations in reverse order
+            for compensation in reversed(compensations):
+                await workflow.execute_activity(
+                    compensation,
+                    input,
+                    start_to_close_timeout=timedelta(seconds=10),
+                )
+
+            # Return a message indicating the booking process was cancelled
+            return "Voyage cancelled"
+```
+
+The `compensations` list keeps track of the actions that need to be undone in case of a failure.
+Each compensation action is appended to this list after its corresponding booking action is successfully completed.
+The `try` block attempts to execute each booking Activity (`book_car`, `book_hotel`, `book_flight`) in sequence.
+Each Activity Execution includes a retry policy to handle transient errors.
+If any Activity fails, the `except` block catches the exception and executes the compensation activities in reverse order to undo the previously completed steps.
+This ensures the system returns to a consistent state. The retry policy specifies how to handle retries for each Activity, including non-retryable error types and retry intervals.
+
+Having defined the Workflow, you're now ready to set up the Worker that will execute these Workflows and Activities.
+
+## Define the Worker
+
+To make your booking logic, built with Temporal Workflows and Activities, functional and integrated into your application, you need to set up a Worker.
```

Comment:
Nit: You could make the Worker more active in this sentence (or sentences). Setting up a Worker lets you integrate... A Worker is built with...